core/templates/domain/topic/editable-topic-backend-api.service.spec.ts

```diff
@@ -224,86 +232,109 @@ describe('Editable topic backend API service', function() {
+    ' topic fails', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
 
-    $httpBackend.expect('GET', '/topic_editor_story_handler/' + '0')
-      .respond(500, 'Error loading story with id 0.');
-    EditableTopicBackendApiService.fetchStories('0').then(
+    editableTopicBackendApiService.fetchStories('0').then(
       successHandler, failHandler);
-    $httpBackend.flush();
+    let req = httpTestingController.expectOne(
+      '/topic_editor_story_handler/' + '0');
+    expect(req.request.method).toEqual('GET');
+    req.flush('Error loading story with id 0.', {
+      status: 500, statusText: 'Invalid Request'
+    });
+
+    flushMicrotasks();
 
     expect(successHandler).not.toHaveBeenCalled();
     expect(failHandler).toHaveBeenCalledWith(
       'Error loading story with id 0.');
-  });
+  }));
 
-  it('should sucessfully delete a topic', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
+  it('should sucessfully delete a topic', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
 
-    $httpBackend.expectDELETE('/topic_editor_handler/data/' + '0')
-      .respond(200);
-    EditableTopicBackendApiService.deleteTopic('0').then(
+    editableTopicBackendApiService.deleteTopic('0').then(
       successHandler, failHandler);
-    $httpBackend.flush();
+    let req = httpTestingController.expectOne(
+      '/topic_editor_handler/data/' + '0');
+    expect(req.request.method).toEqual('DELETE');
+    req.flush('Topic with given id deleted.', {
+      status: 200, statusText: 'Success'
+    });
+
+    flushMicrotasks();
 
     expect(successHandler).toHaveBeenCalledWith(200);
     expect(failHandler).not.toHaveBeenCalled();
-  });
+  }));
 
   it('should use the rejection handler when deleting a topic fails',
-    function() {
-      var successHandler = jasmine.createSpy('success');
-      var failHandler = jasmine.createSpy('fail');
+    fakeAsync(() => {
+      let successHandler = jasmine.createSpy('success');
+      let failHandler = jasmine.createSpy('fail');
 
-      $httpBackend.expectDELETE('/topic_editor_handler/data/' + '1')
-        .respond(500, 'Error deleting topic 1.');
-      EditableTopicBackendApiService.deleteTopic('1').then(
+      editableTopicBackendApiService.deleteTopic('1').then(
         successHandler, failHandler);
-      $httpBackend.flush();
+      let req = httpTestingController.expectOne(
+        '/topic_editor_handler/data/' + '1');
+      expect(req.request.method).toEqual('DELETE');
+      req.flush('Error deleting topic 1.', {
+        status: 500, statusText: 'Invalid Request'
+      });
+
+      flushMicrotasks();
 
       expect(successHandler).not.toHaveBeenCalled();
-      expect(failHandler).toHaveBeenCalledWith('Error deleting topic 1.');
+      expect(failHandler).toHaveBeenCalled();
```

Comment:
Why are we not using `toHaveBeenCalledWith('Error deleting topic 1.')` See Line no. 298

Reply:
One of the earlier changes I made to get my head around the code and missed. Thanks @Showtim3 , fixed it!